File: app/src/main/java/com/coder/ffmpegtest/ui/adapter/FFmpegCommandAdapter.java
package com.coder.ffmpegtest.ui.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.coder.ffmpegtest.R;
import com.coder.ffmpegtest.model.CommandBean;
import com.coder.ffmpegtest.ui.vh.FFmpegCommandViewHolder;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

/**
 * @author: AnJoiner
 * @datetime: 19-12-30
 */
public class FFmpegCommandAdapter extends RecyclerView.Adapter<FFmpegCommandViewHolder> {

    private List<CommandBean> mStrings;
    private ItemClickListener mItemClickListener;

    public void setItemClickListener(ItemClickListener itemClickListener) {
        mItemClickListener = itemClickListener;
    }

    public FFmpegCommandAdapter(List<CommandBean> strings) {
        mStrings = strings;
    }

    @NonNull
    @Override
    public FFmpegCommandViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.adapter_ffmpeg_command,parent,false);
        return new FFmpegCommandViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final FFmpegCommandViewHolder holder, int position) {
        holder.mButton.setText(mStrings.get(position).getName());
        holder.mButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mItemClickListener!=null){
                    mItemClickListener.itemClick(mStrings.get(holder.getAdapterPosition()).getId());
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return mStrings.size();
    }

    public interface ItemClickListener{
        void itemClick(int id);
    }
}

File: app/src/main/java/com/coder/ffmpegcommand/ui/adapter/FFmpegCommandAdapter.kt
package com.coder.ffmpegcommand.ui.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.coder.ffmpegcommand.R
import com.coder.ffmpegcommand.model.CommandBean
import com.coder.ffmpegcommand.ui.vh.FFmpegCommandViewHolder

/**
 * @author: AnJoiner
 * @datetime: 19-12-30
 */
class FFmpegCommandAdapter(private val mStrings: List<CommandBean>) : RecyclerView.Adapter<FFmpegCommandViewHolder>() {
    private var mItemClickListener: ItemClickListener? = null
    
    fun setItemClickListener(itemClickListener: ItemClickListener) {
        mItemClickListener = itemClickListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FFmpegCommandViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.adapter_ffmpeg_command, parent, false)
        return FFmpegCommandViewHolder(view)
    }

    override fun onBindViewHolder(holder: FFmpegCommandViewHolder, position: Int) {
        holder.mButton.text = mStrings[position].name
        holder.mButton.setOnClickListener {
            if (mItemClickListener != null) {
                mItemClickListener!!.itemClick(mStrings[position].id)
            }
        }
    }

    override fun getItemCount(): Int {
        return mStrings.size
    }

    interface ItemClickListener {
        fun itemClick(id: Int)
    }

}